src/section/shared_state.rs

```diff
@@ -0,0 +1,854 @@
+            }
+        };
+
+        if self.is_peer_our_elder(&details.pub_id) {
+            warn!(
+                "Not relocating {} - The peer is still our elder.",
+                details.pub_id,
+            );
+
+            // Keep the details in the queue so when the node is demoted we can relocate it.
+            self.relocate_queue.push_back(details);
+            return None;
+        }
+
+        trace!("relocating member {}", details.pub_id);
+        Some(details)
+    }
+
+    /// Provide a SectionProofSlice that proves the given signature to the given destination
+    /// location.
+    /// If `node_knowledge_override` is `Some`, it is used when calculating proof for
+    /// `DstLocation::Node` instead of the stored knowledge. Has no effect for other location types.
+    pub fn prove(
+        &self,
+        target: &DstLocation,
+        node_knowledge_override: Option<u64>,
+    ) -> SectionProofSlice {
+        let first_index = self
+            .sections
+            .knowledge_index(target, node_knowledge_override);
+        self.our_history.slice_from(first_index as usize)
+    }
+
+    /// Check if we know this node but have not yet processed it.
+    pub fn is_in_online_backlog(&self, pub_id: &PublicId) -> bool {
+        self.churn_event_backlog.iter().any(|evt| {
+            if let AccumulatingEvent::Online(payload) = &evt {
+                payload.p2p_node.public_id() == pub_id
+            } else {
+                false
+            }
+        })
+    }
+
+    // Tries to split our section.
+    // If we have enough mature nodes for both subsections, returns the elders infos of the two
+    // subsections. Otherwise returns `None`.
+    fn try_split(
+        &self,
+        network_params: &NetworkParams,
+        our_name: &XorName,
+    ) -> Result<Option<(EldersInfo, EldersInfo)>> {
+        let next_bit_index = self.our_prefix().bit_count();
+        let next_bit = our_name.bit(next_bit_index);
+
+        let (our_new_size, sibling_new_size) = self
+            .our_members
+            .mature()
+            .map(|p2p_node| p2p_node.name().bit(next_bit_index) == next_bit)
+            .fold((0, 0), |(ours, siblings), is_our_prefix| {
+                if is_our_prefix {
+                    (ours + 1, siblings)
+                } else {
+                    (ours, siblings + 1)
+                }
+            });
+
+        // If either of the two new sections will not contain enough entries, return `false`.
+        if our_new_size < network_params.safe_section_size
+            || sibling_new_size < network_params.safe_section_size
+        {
+            return Ok(None);
+        }
+
+        let our_prefix = self.our_prefix().pushed(next_bit);
+        let other_prefix = self.our_prefix().pushed(!next_bit);
+
+        let our_elders = self
+            .our_members
+            .elder_candidates_matching_prefix(&our_prefix, network_params.elder_size);
+        let other_elders = self
```

Comment:
just to confirm that we don't need to ensure the number of elders within the split out sections, right?

Reply:
Could of lines above is the check that both sub-sections have enough members which should be enough I think.

Reply:
hmm...  what I mean is that the above check only ensure both sections got enough `members`.
However it doesn't check whether the new sections can all get enough qualified elders.

I am not sure it is required to carry out this check (or, if as you mentioned, `members` refer to `edlers` actually).

Reply:
Got you. What matters for the split to happen is only that both sub-sections get enough *members*. The elders are selected only afterwards. There is no such thing as "qualified elder" - any node can become elder if it is among the 7 oldest ones of the section.

Reply:
ok. then.